hadoop-common-project/hadoop-common/src/main/java/io/hops/security/UsersGroups.java

```diff
@@ -15,400 +15,163 @@
-      for (String group : groups) {
-        flushGroup(group);
-      }
-      addUserToGroupsInternalTx(user, groups);
-    } catch (UniqueKeyConstraintViolationException ex){
-      LOG.debug("User/Group was already added: " + ex);
     }
+    usersGroupsMapping.addUserToGroupTx(user, group);
   }
-
-  public static void addUserToGroups(final String user, final String[]
+  
+  public static void addUserToGroupsTx(String user, String[]
       groups) throws IOException {
-
-    if(!isConfigured)
+    if(usersGroupsMapping == null){
+      LOG.warn("UsersGroups was not initialized.");
       return;
-
-    addUserToGroupsInternal(user, groups);
-  }
-
-  private static void addUserToGroupsInternalTx(final String user, final
-  String[] grps) throws IOException {
-      new LightWeightRequestHandler(UsersOperationsType.ADD_USER) {
-        @Override
-        public Object performTask() throws StorageException, IOException {
-          addUserToGroupsInternal(user, grps);
-          return null;
-        }
-      }.handle();
-  }
-
-  private static void addUserToGroupsInternal(final String user, final
-      String[] grps) throws StorageException {
-
-    Collection<String> groups = null;
-
-    if(grps != null) {
-      groups = Collections2.filter(Arrays.asList(grps), Predicates
-          .<String>notNull());
     }
-
-    List<String> availableGroups = cache.getGroups(user);
-    if(availableGroups != null && groups != null){
-      if(availableGroups.containsAll(groups)){
-        return;
-      }
+    usersGroupsMapping.addUserToGroupsTx(user, groups);
+  }
+  
+  public static void addUserToGroup(String user, String group) throws IOException {
+    if(usersGroupsMapping == null){
+      LOG.warn("UsersGroups was not initialized.");
+      return;
     }
-
-    Integer userId = cache.getUserId(user);
-
-    if(userId == null && user != null) {
-      User addedUser = userDataAccess.addUser(user);
-      cache.addUser(addedUser);
-      userId = addedUser.getId();
+    usersGroupsMapping.addUserToGroup(user, group);
+  }
+  
+  public static void addUserToGroups(String user, String[]
+      groups) throws IOException {
+    if(usersGroupsMapping == null){
+      LOG.warn("UsersGroups was not initialized.");
+      return;
     }
-
-    if(groups != null){
-      for(String group : groups){
-        List<Integer> groupIds = Lists.newArrayList();
-        Integer groupId = cache.getGroupId(group);
-        if(groupId == null){
-          Group addedGroup = groupDataAccess.addGroup(group);
-          cache.addGroup(addedGroup);
-          groupId = addedGroup.getId();
-        }
-        groupIds.add(groupId);
-
-        if(userId != null) {
-          userGroupDataAccess.addUserToGroups(userId, groupIds);
-        }
-      }
-
-      if(user != null){
-        cache.appendUserGroups(user, groups);
-      }
+    usersGroupsMapping.addUserToGroups(user, groups);
+  }
+  
+  public static void flushCache(String user, String group){
```

Comment:
When flushing shouldn't it also persist everything in DB instead of just invalidating the cache

Reply:
Nope, the method name is misleading, i forgot to change the name for that method. It should only delete the entries from the cache.